crates/table/src/table.rs

```diff
@@ -486,10 +486,12 @@ impl Table {
     /// On `Ok(_)`, statistics of the table are also updated,
     /// and the `ptr` still points to a valid row, and otherwise not.
     ///
+    /// If `CHECK_SAME_ROW` holds, an identical row will be treated as a set-semantic duplicate.
+    ///
     /// # Safety
     ///
     /// `self.is_row_present(row)` must hold.
-    pub unsafe fn confirm_insertion<'a>(
+    pub unsafe fn confirm_insertion<'a, const CHECK_SAME_ROW: bool>(
```

Comment:
I kind of hate these const bool parameters, tbh. A couple alternatives that make the callsites more readable:

- Have a private "inner" function which takes the flag, and two different "outer" functions with descriptive names. E.g. `confirm_insertion_check_same_row` and `confirm_insertion_ignore_same_row`.
- Define a custom enum with two variants with descriptive names. E.g. `SameRow::Reject` and `SameRow::Ignore`.
- Add a comment to each callsite which says what the flag does and why the chosen value is correct. E.g.
  ```rust
    // `CHECK_SAME_ROW = false` because we're checking the row against the table we just inserted it into,
    // so obviously it will be present.
    ````